File: mobile/src/types/house.ts
export type DealType = 'JEONSE' | 'WOLSE' | 'BAN_JEONSE';

export interface Address {
  roadAddress: string;
  jibunAddress: string;
  zonecode: string;
  latitude?: number;
  longitude?: number;
  detail?: string;
}

export const RATING_KEYS = [
  'waterPressure',
  'sunlight',
  'noise',
  'insulation',
  'ventilation',
  'moisture',
  'neighborhood',
  'firstImpression',
] as const;
export type RatingKey = (typeof RATING_KEYS)[number];

// 방 구조 — DB에는 영문 코드 저장, 라벨 매핑은 UI에서.
export const ROOM_TYPES = [
  'ONE_ROOM', // 원룸
  'SEPARATED_ONE_ROOM', // 분리형 원룸
  'ONE_AND_HALF', // 1.5룸
  'TWO_ROOM', // 투룸
  'THREE_ROOM_PLUS', // 쓰리룸+
] as const;
export type RoomType = (typeof ROOM_TYPES)[number];

// 층 유형 — 지상 / 반지하 / 옥탑
export const FLOOR_TYPES = ['GROUND', 'SEMI_BASEMENT', 'ROOFTOP'] as const;
export type FloorType = (typeof FLOOR_TYPES)[number];

// 향 — 남/남동/남서/동/서/북
export const DIRECTIONS = [
  'SOUTH',
  'SOUTHEAST',
  'SOUTHWEST',
  'EAST',
  'WEST',
  'NORTH',
] as const;
export type Direction = (typeof DIRECTIONS)[number];

// 관리비 포함 가능 항목 — 수도/전기/가스/인터넷
export const MAINTENANCE_UTILITIES = ['WATER', 'ELECTRIC', 'GAS', 'INTERNET'] as const;
export type MaintenanceUtility = (typeof MAINTENANCE_UTILITIES)[number];

/** 컨디션 3단계: 1=나쁨, 2=보통, 3=좋음 */
export type ConditionLevel = 1 | 2 | 3;

export interface House {
  id: string;
  address: Address;
  dealType: DealType;
  deposit: number;
  rent?: number;
  maintenanceFee?: number;
  area?: number; // 단위: 평(pyeong)
  builtYear?: number;
  floor?: number;
  totalFloor?: number;
  availableFrom?: string;
  stationDistance?: number;
  rooms?: number;
  bathrooms?: number;
  hasBalcony?: boolean;
  hasElevator?: boolean;
  hasParking?: boolean;
  options?: string[];
  security?: string[];
  garbage?: string;
  waterPressure?: number;
  sunlight?: number;
  noise?: number;
  insulation?: number;
  ventilation?: number;
  moisture?: number;
  neighborhood?: number;
  firstImpression?: number;
  memo?: string;
  // ── 집 추가 위저드 신규 필드 ──
  nickname?: string; // 집 별칭 (위저드에서 필수, 타입상 옵셔널)
  visitedAt?: string; // 방문일 (ISO), 기본 오늘
  roomType?: RoomType; // 방 구조 (숫자 rooms와 별개)
  floorType?: FloorType; // 지상/반지하/옥탑
  direction?: Direction; // 향
  maintenanceIncludes?: MaintenanceUtility[]; // 관리비 포함 항목
  utilityEstimates?: Partial<Record<'WATER' | 'ELECTRIC' | 'GAS', number>>; // 별도 납부 월 예상(만원)
  fullOption?: boolean; // 풀옵션(빌트인)
  photoIds: string[];
  contractedAt?: string;
  createdAt: string;
  updatedAt: string;
}

export type HouseDraft = Omit<House, 'id' | 'photoIds' | 'createdAt' | 'updatedAt'>;

export type HouseRequestFields = Omit<House, 'id' | 'photoIds' | 'createdAt' | 'updatedAt'>;

export type CreateHouseRequest = Omit<HouseRequestFields, 'rent'> & {
  rent: number;
};

export type UpdateHouseRequest = Partial<CreateHouseRequest>;

File: backend/src/main/java/app/pickhouse/domain/house/entity/House.java
package app.pickhouse.domain.house.entity;

import app.pickhouse.global.address.Address;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.LocalDate;
import java.util.UUID;

@Entity
@Table(name = "houses")
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder(toBuilder = true)
public class House {

    @Id @Column(columnDefinition = "CHAR(36)") private UUID id;
    @Column(name = "user_id", nullable = false, columnDefinition = "CHAR(36)") private UUID userId;

    @Embedded
    @AttributeOverrides({
        @AttributeOverride(name = "roadAddress",  column = @Column(name = "address_road_address")),
        @AttributeOverride(name = "jibunAddress", column = @Column(name = "address_jibun_address")),
        @AttributeOverride(name = "zonecode",     column = @Column(name = "address_zonecode")),
        @AttributeOverride(name = "latitude",     column = @Column(name = "address_latitude")),
        @AttributeOverride(name = "longitude",    column = @Column(name = "address_longitude")),
        @AttributeOverride(name = "detail",       column = @Column(name = "address_detail"))
    })
    private Address address;

    @Enumerated(EnumType.STRING)
    @Column(name = "deal_type", nullable = false)
    private DealType dealType;

    @Column(nullable = false) private int deposit;
    @Column(nullable = false) private int rent;
    @Column(name = "maintenance_fee") private Integer maintenanceFee;
    @Column private BigDecimal area;
    @Column(name = "built_year") private Integer builtYear;
    @Column private Integer floor;
    @Column(name = "total_floor") private Integer totalFloor;
    @Column(name = "available_from") private LocalDate availableFrom;
    @Column(name = "station_distance") private Integer stationDistance;

    @Column private Integer rooms;
    @Column private Integer bathrooms;
    @Column(name = "has_balcony") private Boolean hasBalcony;
    @Column(name = "has_elevator") private Boolean hasElevator;
    @Column(name = "has_parking") private Boolean hasParking;

    @Column(name = "options_json", columnDefinition = "json") private String optionsJson;
    @Column(name = "security_json", columnDefinition = "json") private String securityJson;
    @Column private String garbage;

    @JdbcTypeCode(SqlTypes.TINYINT) @Column(name = "water_pressure") private Integer waterPressure;
    @JdbcTypeCode(SqlTypes.TINYINT) @Column private Integer sunlight;
    @JdbcTypeCode(SqlTypes.TINYINT) @Column private Integer noise;
    @JdbcTypeCode(SqlTypes.TINYINT) @Column private Integer insulation;
    @JdbcTypeCode(SqlTypes.TINYINT) @Column private Integer ventilation;
    @JdbcTypeCode(SqlTypes.TINYINT) @Column private Integer moisture;
    @JdbcTypeCode(SqlTypes.TINYINT) @Column private Integer neighborhood;
    @JdbcTypeCode(SqlTypes.TINYINT) @Column(name = "first_impression") private Integer firstImpression;

    @Column(columnDefinition = "TEXT") private String memo;

    // ── 집 추가 위저드 신규 필드 ──
    @Column(length = 100) private String nickname;
    @Column(name = "visited_at") private Instant visitedAt;
    @Column(name = "contracted_at") private Instant contractedAt;

    @Enumerated(EnumType.STRING)
    @Column(name = "room_type")
    private RoomType roomType;

    @Enumerated(EnumType.STRING)
    @Column(name = "floor_type")
    private FloorType floorType;

    @Enumerated(EnumType.STRING)
    @Column
    private Direction direction;

    @Column(name = "maintenance_includes_json", columnDefinition = "json") private String maintenanceIncludesJson;
    @Column(name = "utility_estimates_json", columnDefinition = "json") private String utilityEstimatesJson;
    @Column(name = "full_option") private Boolean fullOption;

    @Column(name = "created_at", nullable = false) private Instant createdAt;
    @Column(name = "updated_at", nullable = false) private Instant updatedAt;
    @Column(name = "deleted_at") private Instant deletedAt;

    public void softDelete(Instant now) { this.deletedAt = now; this.updatedAt = now; }
    public void touch(Instant now) { this.updatedAt = now; }
}
